src/script/interpreter.cpp

```diff
@@ -1427,12 +1529,19 @@ void PrecomputedTransactionData::Init(const T& txTo, std::vector<CTxOut>&& spent
         if (uses_bip341_taproot && uses_bip143_segwit) break; // No need to scan further if we already need all.
     }
 
```

Comment:
Computation required by CTV ideally would not be carried out when CTV could not have been active.
In this context the `tx.version` field is available. If CTV required version 3 or higher (not sure this is a good idea or not?), then it would be trivial to augment the old logic that was more conservative with resource use. (again not sure how much that matters in practice?)